etcdmain/etcd.go

```diff
@@ -160,6 +160,12 @@ func Main() {
 				}
 				os.Exit(1)
 			}
+			if etcdserver.IsDiscoveryError(err) {
+				plog.Errorf("%v", err)
+				plog.Infof("Interaction with discovery service failed, possibly due to connection down.")
```

Comment:
I think all we need to say here is that this discovery failed and you have to generate a new token rather than reuse the old one.